### PR TITLE
feat: DAH-2878 sign out connection issue

### PR DESCRIPTION
**Pull request by tallulahkay**
## Description

Add an alert for when a sign out is triggered by an expired token.

## Jira ticket

https://sfgovdt.jira.com/browse/DAH-2878

### Version Control

- [ ] branch name begins with `angular` if it contains updates to Angular code
- [x] branch name contains the Jira ticket number
- [x] PR name follows `type: TICKET-NUMBER Description` format, e.g. `feat: DAH-123 New Feature`

### Code quality

- [x] [the set of changes is small](https://google.github.io/eng-practices/review/developer/small-cls.html#what-is-small)
- [x] all automated code checks pass (linting, tests, coverage, etc.)
- [x] code irrelevant to the ticket is not modified e.g. changing indentation due to automated formatting
- [x] if the code changes the UI, it matches the UI design exactly
- [ ] if the changes include human translations, follow the [human translations process](https://sfgovdt.jira.com/l/cp/XS1KpvE4)

### Review instructions

- [x] instructions specify which environment(s) it applies to
- [x] instructions work for PA testers
- [x] instructions have already been performed at least once

### Request review

- [x] PR has `needs review` label
- [x] Use `Housing Eng` group to automatically assign reviewers, and/or assign specific engineers
- [ ] If time sensitive, notify engineers in Slack

### CODE REVIEW

File: app/javascript/authentication/token.ts [hunk 1]
```diff
@@ -63,6 +63,13 @@ export const clearHeadersTimeOut = () => {
   getStorage().removeItem(ACCESS_TOKEN_LOCAL_STORAGE_KEY)
 }
 
+export const clearHeadersConnectionIssue = () => {
+  if (getStorage().getItem(ACCESS_TOKEN_LOCAL_STORAGE_KEY)) {
+    getStorage().setItem("alert_message_secondary", t("signOut.alertMessage.connectionIssue"))
+    getStorage().removeItem(ACCESS_TOKEN_LOCAL_STORAGE_KEY)
+  }
```
Comment (jimlin-sfgov) [inline app/javascript/authentication/token.ts:70]: question: Why do the other `clear...` functions always remove the token (outside of the `if` statement), while this one does not?

Comment (tallulahkay) [reply app/javascript/authentication/token.ts:70]: No reason - should function effectively the same. I can change this one for consistency though!

File: app/javascript/authentication/token.ts [hunk 2]
```diff
@@ -57,7 +56,14 @@ export const clearHeadersSignOut = () => {
...
     setSiteAlertMessage(t("signOut.alertMessage.timeout"), "secondary")
-    getStorage().removeItem(ACCESS_TOKEN_LOCAL_STORAGE_KEY)
+  }
+
+  getStorage().removeItem(ACCESS_TOKEN_LOCAL_STORAGE_KEY)
+}
+
+export const clearHeadersConnectionIssue = () => {
+  if (getStorage().getItem(ACCESS_TOKEN_LOCAL_STORAGE_KEY)) {
+    getStorage().setItem("alert_message_secondary", t("signOut.alertMessage.connectionIssue"))
```
Comment (jimlin-sfgov) [inline app/javascript/authentication/token.ts:66]: suggestion: Use the `setSiteAlertMessage` function for consistency.

**Review by jimlin-sfgov — APPROVED** [review]
LGTM with a non-blocking suggestion

**Pull request merged**